### DISCUSSION TITLE
Multiple gNBs on ZMQ set up

### DISCUSSION THREAD

**Discussion by christsaks** [srsran/srsRAN_Project#561] [General Help] [answered]
Hi to all,
I am trying to run 2 gnBs using the ZMQ set up but I am facing some issues. I know there is much discussion on multi-UE support and that is possible with GNU radio and i have successfully set up this but with multi gnb i am facing some challenges. Are there changes needed to the open5gs side to support this? Is there some tutorial for this set up?  I have observed that using a different GTPU_PORT in the second gnb instance makes it connect to the open5gs but problems come later when i try to connect a UE to that gnB. Maybe some change is needed in the conf files of the open5gs?
Thanks in advance for any help

**Comment by pgawlowicz** [comment, marked as answer]
To connect the 2nd gNB to the same Open5gs running in docker, one need to add a second IP address to the bridge interface connecting to the Open5gs docker container.

1. Get the name of the Open5gs docker bridge:
```
ip -o addr show | grep "10.53.1.1"
```
2. Add a second IP address to the bridge:
```
sudo ip addr add 10.53.1.3/24 dev BRIDGE_NAME
```
3. Use the `IP=10.53.1.3` as `amf.bind_addr` in the 2nd gNB config.

4. Remember to use different ZMQ ports for the gNB-UE pairs. Also, the UE has to have different IMSI, that need to be added to the Open5gs database.

You can take the Open5gs subscriber database and the UE configs from our [tutorial](https://docs.srsran.com/projects/project/en/latest/tutorials/source/srsUE/source/index.html#multi-ue-emulation). Remember to adapt ZMQ ports!

**Reply by christsaks** [reply]
Thank you! Following you instructions i was able to connect 2 gnbs and ping  to their UEs!

**Reply by newt2105** [reply]
> To connect the 2nd gNB to the same Open5gs running in docker, one need to add a second IP address to the bridge interface connecting to the Open5gs docker container.
> 
> 1. Get the name of the Open5gs docker bridge:
> 
> ```
> ip -o addr show | grep "10.53.1.1"
> ```
> 
> 2. Add a second IP address to the bridge:
> 
> ```
> sudo ip addr add 10.53.1.3/24 dev BRIDGE_NAME
> ```
> 
> 3. Use the `IP=10.53.1.3` as `amf.bind_addr` in the 2nd gNB config.
> 4. Remember to use different ZMQ ports for the gNB-UE pairs. Also, the UE has to have different IMSI, that need to be added to the Open5gs database.
> 
> You can take the Open5gs subscriber database and the UE configs from our [tutorial](https://docs.srsran.com/projects/project/en/latest/tutorials/source/srsUE/source/index.html#multi-ue-emulation). Remember to adapt ZMQ ports!

Hi @pgawlowicz, 
Thanks about your instruction, I did it
I have one more question: how about connect multiple gnb with RIC, Is it have same way?
Thanks a lot!

**Reply by ethanlin355** [reply]
> > To connect the 2nd gNB to the same Open5gs running in docker, one need to add a second IP address to the bridge interface connecting to the Open5gs docker container.
> > 
> > 1. Get the name of the Open5gs docker bridge:
> > 
> > ```
> > ip -o addr show | grep "10.53.1.1"
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > 2. Add a second IP address to the bridge:
> > 
> > ```
> > sudo ip addr add 10.53.1.3/24 dev BRIDGE_NAME
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > 3. Use the `IP=10.53.1.3` as `amf.bind_addr` in the 2nd gNB config.
> > 4. Remember to use different ZMQ ports for the gNB-UE pairs. Also, the UE has to have different IMSI, that need to be added to the Open5gs database.
> > 
> > You can take the Open5gs subscriber database and the UE configs from our [tutorial](https://docs.srsran.com/projects/project/en/latest/tutorials/source/srsUE/source/index.html#multi-ue-emulation). Remember to adapt ZMQ ports!
> 
> Hi @pgawlowicz, Thanks about your instruction, I did it I have one more question: how about connect multiple gnb with RIC, Is it have same way? Thanks a lot!

It is the same way just use the ric subnet is fine
